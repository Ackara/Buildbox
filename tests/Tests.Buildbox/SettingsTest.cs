using Ackara.Buildbox.SemVer;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Shouldly;
using System.IO;

namespace Tests.Buildbox
{
    [TestClass]
    [UseApprovalSubdirectory(nameof(ApprovalTests))]
    [UseReporter(typeof(FileLauncherReporter), typeof(ClipboardReporter))]
    public class SettingsTest
    {
        [TestMethod]
        public void Load_should_generate_a_default_settings_file_when_it_do_not_exist()
        {
            // Act
            if (File.Exists(Settings.DefaultSettingsPath))
            {
                File.Delete(Settings.DefaultSettingsPath);
            }

            var result = Settings.Load();
            bool settingsFileWasCreated = File.Exists(Settings.DefaultSettingsPath);

            // Assert
            result.ShouldNotBeNull();
            settingsFileWasCreated.ShouldBeTrue();
            Approvals.VerifyFile(Settings.DefaultSettingsPath);
        }
    }
}