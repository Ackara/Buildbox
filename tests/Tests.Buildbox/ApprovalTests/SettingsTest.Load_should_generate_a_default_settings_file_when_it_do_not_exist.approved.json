{
  "version": {
    "major": 0,
    "minor": 0,
    "patch": 0,
    "releaseTag": ""
  },
  "targets": [
    "dotnetcore",
    "dotnet",
    "powershell"
  ],
  "releaseTags": {
    "master": "",
    "*": "-alpha"
  },
  "shouldCommitChanges": true,
  "shouldTagCommit": true,
  "shouldAddAllUnstagedFilesWhenCommitting": false
}